src/t2iapi/context/service.proto

```diff
@@ -158,6 +158,52 @@ service ContextService {
   rpc EnsembleContextIndicateMembershipWithIdentification (BasicHandleRequest)
       returns (EnsembleContextIndicateMembershipWithIdentificationResponse);
 
+  /*
+  Indicate membership in an SDC PARTICIPANT ENSEMBLE resp. in SDC PARTICIPANT ENSEMBLEs for a provided
+  pm:EnsembleContextDescriptor by associating each state contained in a representative set of pm:EnsembleContextStates
+  of that descriptor. Return the list of handles of states contained in that representative set.
+
+  The representative set of pm:EnsembleContextStates for a given pm:EnsembleContextDescriptor is a minimal set of
+  pm:EnsembleContextStates which if associated are sufficient to indicate membership in the SDC PARTICIPANT ENSEMBLE(s)
+  for that descriptor.
+
+  Example 1: treatment session
+    A POC MEDICAL DEVICE currently working in a (temporal) logical group based on a treatment session
+    may expressed by a pm:EnsembleContextDescriptor. If each of its pm:EnsembleContextStates describes a step in
+    this treatment session, then it is sufficient if the manipulation returns only one state handle. It is
+    representative since all pm:EnsembleContextStates of the pm:EnsembleContextDescriptor behave the same and indicate
+    membership in the logical group resp. SDC PARTICIPANT ENSEMBLE if associated.
+
+  Example 2: location inferring
+    If a pm:EnsembleContextDescriptor has multiple pm:EnsembleContextStates whereby each describes a membership of a
+    location-inferred SDC PARTICIPANT ENSEMBLE if associated. It is sufficient that the manipulation returns only one
+    state handle since it is representative. All pm:EnsembleContextStates of the pm:EnsembleContextDescriptor behave the
+    same independent of the location allocation.
+
+  Example 3: general hospital process
+    Consider a general process within the hospital with multiple steps whereby each step is divided into multiple minor
+    steps describing the major step more precisely. Membership of this main process as well as of the sub processes is
+    modeled by a pm:EnsembleContextDescriptor that has multiple pm:EnsembleContextStates for the major as well as for
+    the minor steps. To indicate membership in the main as well as in the sub process a
+    pm:EnsembleContextState representing a major step and a pm:EnsembleContextState representing a major's minor step
+    have to be associated. It is sufficient that the manipulation returns the state handles of these two
+    pm:EnsembleContextStates.
+
```

Comment:
Just something that is currently unclear:
What happens when the nesting depth is not constant? For instance, when some major steps do not have minor steps or when some minor steps have even smaller substeps?

Is it then sufficient to choose a major step that does not have minor steps (one EnsembleContextState) or should one rather maximize the number of EnsembleContextStates?

Reply:
I have assumed that the process steps are structurally the same. Of course, that doesn't have to be the case. In that case, I would probably expect that the set of representative states has to be extended. I adapted the description.